alchemist/alchemist-ui-tooling/build.gradle.kts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2010-2019) Danilo Pianini and contributors listed in the main project"s alchemist/build.gradle file.
+ *
+ * This file is part of Alchemist) and is distributed under the terms of the
+ * GNU General Public License) with a linking exception)
+ * as described in the file LICENSE in the Alchemist distribution"s top directory.
+ */
+
+dependencies {
+    implementation(rootProject)
+    implementation(project(":alchemist-implementationbase"))
+    implementation(project(":alchemist-euclidean-geometry"))
+    implementation(project(":alchemist-maps"))
+    val javaFXVersion = "11"
+    for (platform in listOf("linux", "mac", "win")) {
```

Comment:
Tooling in there should be FX-agnostic. This stuff is reusable elsewhere

Reply:
Also I don't want to pull in JFX if I'm using swing ui

Reply:
JavaFX was used by Wormhole2D because it used to have one constructor for JavaFX and one for SwingUI.

[This](https://github.com/Vuksaa/Alchemist/blob/develop/alchemist/alchemist-ui-tooling/src/main/java/it/unibo/alchemist/boundary/wormhole/implementation/Wormhole2D.java#L27) is my attempt to make the constructor implementation-independent.
Now I feel like it would make sense to make Wormhole2D abstract and create concrete classes for JavaFX and Swing, since otherwise they have to specify the ViewType -> PointAdapter converter every time a Wormhole2D is instanced. Should I proceed with this?

Reply:
In general, yes, having an abstract class in the tooling project and then two implementations one for Swing and one for JFX is the way to go.
However, it seems to me that `AbstractWormhole2D` is abstract already, so I don't fully understand what you mean by "make Wormhole2D abstract"

Reply:
Ah, I had forgotten about AbstractWormhole2D.
Then I would merge `Wormhole2D` into `AbstractWormhole2D` instead and create the implementations `WormholeSwing` and `WormholeFX` _(I do not like these names..)_.
At that point, [MapWormhole](https://github.com/Vuksaa/Alchemist/blob/develop/alchemist/alchemist-swingui/src/main/java/it/unibo/alchemist/boundary/wormhole/implementation/MapWormhole.java) and [LeafletMapWormhole](https://github.com/Vuksaa/Alchemist/blob/develop/alchemist/alchemist-fxui/src/main/kotlin/it/unibo/alchemist/wormhole/implementation/LeafletMapWormhole.kt) would inherit these implementations.

Reply:
Wait a moment: I'd like `Wormhole2D` to remain an interface. It's ok to have an abstract implementation and two (or more) concrete instances

Reply:
`Wormhole2D` is actually an implementation.

This is the structure on feature-fxui:
`BidimensionalWormhole` (Interface) -> `AbstractWormhole2D` (Abstract) -> `Wormhole2D` (Implementation, JFX/Swing) -> ...

And on develop:
`IWormhole2D` (Interface) -> `Wormhole2D` (Implementation, Swing only) -> ...

So my idea was to do:
`BidimensionalWormhole` (Interface) -> `AbstractWormhole2D` (Abstract)
`AbstractWormhole2D` (Abstract) -> `WormholeFX` (Implementation, JFX) -> ...
`AbstractWormhole2D` (Abstract) -> `WormholeSwing` (Implementation, Swing) -> ...

Reply:
Oh my god, we are in a messing naming situation.
Your proposal is fine. Once done, maybe refactor `BidimensionalWormhole` to `Wormhole2D`